lib/charms/opensearch/v0/opensearch_config.py

```diff
@@ -171,6 +171,11 @@ def set_node(
                 self.CONFIG_YML, "network.publish_host", self._opensearch.host
             )
 
+        if self._opensearch.public_address:
+            self._opensearch.config.put(
+                self.CONFIG_YML, "http.publish_host", self._opensearch.public_address
+            )
```

Comment:
to avoid a double DNS querying, perhaps:
 ```suggestion
        if public_address := self._opensearch.public_address:
            self._opensearch.config.put(
                self.CONFIG_YML, "http.publish_host", public_address
            )
```

Reply:
That if was removed as the idea for the `public_address` to always return an address.